sdk/src/client/api/block_builder/input_selection/remainder.rs

```diff
@@ -110,15 +110,14 @@ impl InputSelection {
 
         let native_tokens_diff = get_native_tokens_diff(&input_native_tokens, &output_native_tokens)?;
 
-        let mut input_mana = 0;
+        let mut input_mana = self.mana_rewards;
```

Comment:
Isn't that dependent on the associated inputs to actually be burnt? Why do we add it always?

Reply:
No, the mana_rewards value is the value of rewards you are trying to claim

Reply:
But ISA doesn't know which output it has to select for this to work, so it only works if you have luck (because it selected the input that makes the rewards available) or if you provide these inputs as required inputs?

Reply:
Yeah but actually I think it's fine because she only add rewards if the delegations are in custom/mandatory inputs, right Chloé?

Reply:
indeed

Reply:
actually, no, it just gets added always 😆